### ISSUE TITLE
在文件内容没有变化的情况下，每次打包出来的构建产物javascript文件的hash值都不一致

### ISSUE THREAD

**Issue by DecadeFarewell** [nuxt/nuxt#27276]
### Environment

Operating System: Linux
Node Version: v20.13.0
Nuxt Version: ^3.11.1
CLI Version: 3.11.1
Nitro Version: 2.9.5
Package Manager: npm@10.5.2
Builder: -
Build Modules: ["@pinia/nuxt", "@element-plus/nuxt", "@vite-pwa/nuxt"]

### Reproduction

- 

### Describe the bug

在文件内容没有变化的情况下，每次执行`nuxt build`,打包出来的js文件hash值都不一致，这导致我在使用docker进行多实例部署时，每个镜像中的构建产物名称不同，用户在请求js资源时，部分资源出现请求404的情况。

<img width="982" alt="企业微信截图_520b0515-691e-4383-904d-6f95669464ab" src="https://github.com/nuxt/nuxt/assets/39828701/be384327-c93a-4ea9-8c90-9280238ba4b8">

![企业微信截图_581c9e05-599b-4f42-a1db-565f3446d589](https://github.com/nuxt/nuxt/assets/39828701/f64f6718-6007-45c1-8ca7-839768968b93)

**Comment by DecadeFarewell** [comment]
现在我可以通过更改构建和发布流程来解决这个问题，我必须在jenkins流水线中完成项目的打包流程，然后在docker镜像启动时运行命令:

**Comment by markthree** [comment]
This seems to be a [rollup](https://github.com/rollup/rollup) problem, for which there is no good solution at the moment.

But maybe we can work around that with [nuxt-build-cache](https://github.com/pi0/nuxt-build-cache).

**Comment by warflash** [comment]
https://github.com/nuxt/nuxt/pull/27258 looks like your issue might be solved by this?

**Comment by yxw007** [comment]
> 现在我可以通过更改构建和发布流程来解决这个问题，我必须在jenkins流水线中完成项目的打包流程，然后在docker镜像启动时运行命令: ` CMD ["npm", "run", "app"]`, 但是这样我无法在docker镜像中注入相关的环境变量以供构建过程中使用，因为构建过程已经结束了
> 
> package.json: ... "scripts": { "build": "nuxt build", "dev": "nuxt dev --host", "start": "node ./.output/server/index.mjs", "app": "pm2 start npm --no-daemon --name portal -- run start", "generate": "nuxt generate", "preview": "nuxt preview", "postinstall": "nuxt prepare" },

告诉你一个解决方案
1. jenkins 打包时，有个选项：将环境变量注入至构建过程（配置环境变量）
2. 在jenkins 打包前执行一个shell，将注入的环境变量生成一个.env
3. docker 打包项目时，将此.env copy 进app中
4. docker 运行时加载.env 配置文件即可（在Dockerfile 中添加如下启动命令）
```shell
# lock the startup env config file 
CMD ["npm","start","--env-file",".env"]
```

**Comment by danielroe** [comment]
As @warflash spotted, this issue is resolved in https://github.com/nuxt/nuxt/pull/27258 in the [nightly release](https://nuxt.com/docs/guide/going-further/nightly-release-channel#nightly-release-channel), or in the next release.

Let me know if not and I'll reopen.

**Comment by DecadeFarewell** [comment]
> As @warflash spotted, this issue is resolved in #27258 in the [nightly release](https://nuxt.com/docs/guide/going-further/nightly-release-channel#nightly-release-channel), or in the next release.
> 
> Let me know if not and I'll reopen.

Okay, thank you. I'll give it a try
